docs/architecture/session.md

```diff
@@ -0,0 +1,66 @@
+# Session
+
+__Session__ is the actor that encapsulates the logic of the sessions 
+associated to each TCP connections.
+
+
+
+## Actor creation and registration
+
+The creation of the session actor and its registration into the system registry are
+performed by the `connection_manager` process:
+
+```rust
+Session::create(move |ctx| {
+    // Get peer address
+    let address = stream.peer_addr().unwrap();
+
+    // Split TCP stream into read and write parts
+    let (r, w) = stream.split();
+
+    // Add stream in session actor from the read part of the tcp stream
+    Session::add_stream(FramedRead::new(r, P2PCodec), ctx);
+
+    // Create the session actor and store in its state the write part of the tcp stream
+    Session::new(address, session_type, FramedWrite::new(w, P2PCodec, ctx))
+});
+```
+
+
+## API
+
+### Outgoing messages: Sessions Manager -> Others
+
+These are the messages sent by the connections manager:
+
+| Message               | Destination             | Input type                                        | Output type                       | Description                           |
+|-----------------------|-------------------------|---------------------------------------------------|-----------------------------------|---------------------------------------|
+| Register              | SessionsManager         | `SocketAddr, Addr<Session>, SessionType`          | `SessionsResult<()>`              | Request to register a new session     |
+| Unregister            | SessionsManager         | `SocketAddr, SessionType`                         | `SessionsResult<()>`              | Request to unregister a session       |
+
+
+#### Register
+
+This message is sent to the [`Sessions Manager`][sessions_manager] actor when the session 
+actor is started to register this session in the hash map of 
```

Comment:
This actor shouldn't care about the specific data structure that `SessionsManager` is using internally for storing references to sessions, and so shouldn't its documentation.